src/backend/distributed/utils/attribute.c

```diff
@@ -0,0 +1,761 @@
+/*-------------------------------------------------------------------------
+ *
+ * attribute.c
+ *	  Routines related to the multi tenant monitor.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "unistd.h"
+
+#include "distributed/citus_safe_lib.h"
+#include "distributed/log_utils.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/jsonbutils.h"
+#include "distributed/colocation_utils.h"
+#include "distributed/tuplestore.h"
+#include "distributed/colocation_utils.h"
+#include "executor/execdesc.h"
+#include "storage/ipc.h"
+#include "storage/lwlock.h"
+#include "storage/shmem.h"
+#include <sys/time.h>
+#include "utils/builtins.h"
+#include "utils/json.h"
+#include "distributed/utils/attribute.h"
+
+
+#include <time.h>
+
+static void AttributeMetricsIfApplicable(void);
+
+ExecutorEnd_hook_type prev_ExecutorEnd = NULL;
+
+#define ATTRIBUTE_PREFIX "/*{\"tId\":"
+#define ATTRIBUTE_STRING_FORMAT "/*{\"tId\":%s,\"cId\":%d}*/"
+#define CITUS_STATS_TENANTS_COLUMNS 7
+#define ONE_QUERY_SCORE 1000000000
+
+/* TODO maybe needs to be a stack */
+char attributeToTenant[MAX_TENANT_ATTRIBUTE_LENGTH] = "";
+CmdType attributeToCommandType = CMD_UNKNOWN;
+int attributeToColocationGroupId = INVALID_COLOCATION_ID;
+
+const char *SharedMemoryNameForMultiTenantMonitor =
+	"Shared memory for multi tenant monitor";
+
+char *tenantTrancheName = "Tenant Tranche";
+char *monitorTrancheName = "Multi Tenant Monitor Tranche";
```

Comment:
These could all be `static` afaict, because they are not used outside of this file. Also in general we try to use `CapitalizedCamelCase` for global variables to make it clear that they are special.
```suggestion
char attributeToTenant[MAX_TENANT_ATTRIBUTE_LENGTH] = "";
CmdType attributeToCommandType = CMD_UNKNOWN;
int attributeToColocationGroupId = INVALID_COLOCATION_ID;

const char *SharedMemoryNameForMultiTenantMonitor =
	"Shared memory for multi tenant monitor";

char *tenantTrancheName = "Tenant Tranche";
char *monitorTrancheName = "Multi Tenant Monitor Tranche";
```